src/main/java/org/jabref/logic/bibtexkeypattern/BibtexKeyGenerator.java

```diff
@@ -72,15 +72,15 @@ private static String getAppendix(int number) {
         }
     }
 
-    public static String cleanKey(String key, boolean enforceLegalKey) {
+    public static String filterUnwantedCharacters(String key, boolean enforceLegalKey) {
```

Comment:
maybe a slightly better name `removeUnwantedCharacters` (filter would imply that the unwanted characters are returned).